src/components/ui.js

```diff
@@ -550,27 +555,14 @@ AFRAME.registerComponent('ui', {
       material.needsUpdate = true;
     });
 
+    this.el.setAttribute('animation__showmessage', { dur: 500, property: 'ui.opacity', from: 0, to: 1, startEvents: 'showmessage' });
+    this.el.setAttribute('animation__hidemessage', { dur: 500, delay: 3000, property: 'ui.opacity', from: 1, to: 0, startEvents: 'hidemessage' });
+    this.el.addEventListener('animationcomplete__showmessage', function (evt) {
```

Comment:
perhaps this one is not needed and we can listen to the event directly in the animation component

```
  this.el.setAttribute('animation__hidemessage', { dur: 500, delay: 3000, property: 'ui.opacity', from: 1, to: 0, startEvents: 'animationcomplete__showmessage' });

```